From be5f55e33f502a8f82aacff6c62964a701f50b3a Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 20 May 2024 03:55:36 +0300
Subject: [PATCH] [SW-15774] board: fix the arc directory

The ARC/ARM build directory is still done in separate directories.
e.g; gaudi3 & gaudi3_arc. The arc image directory still need to be
fixed.

Change-Id: Id14403622705ea00a2106216ad66ba5f490d7d51
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-exports.sh b/board/habanalabs/common/post-image-exports.sh
index 1c1e0d5..bf3f8f6 100755
--- a/board/habanalabs/common/post-image-exports.sh
+++ b/board/habanalabs/common/post-image-exports.sh
@@ -15,5 +15,7 @@
 ITS_SED_PARAMS+="s/@HL_SEC_FW_VERSION@/${HL_SEC_FW_VERSION}/g;";
 ITS_SED_PARAMS+="s/@SHA_VERSION@/${SHA_VERSION}/g;"
 ITS_SED_PARAMS+="s#@HL_UBOOT_BIN_FILE@#${HL_UBOOT_BIN_FILE}#g;"
-ITS_SED_PARAMS+="s#@FILES_PATH_1@#${BINS_DIR}#g;"
-ITS_SED_PARAMS+="s#@FILES_PATH_2@#${BINS_PROD_DIR}#g;"
+ITS_SED_PARAMS+="s#@FILES_PATH_1@#${BINS_PROD_DIR}#g;"
+ITS_SED_PARAMS+="s#@FILES_PATH_2@#${BINS_ARC_DIR}#g;"
+
+export ITS_SED_PARAMS
diff --git a/board/habanalabs/common/post-image-fit-arc.sh b/board/habanalabs/common/post-image-fit-arc.sh
index dcd5606..c7c5d6b 100755
--- a/board/habanalabs/common/post-image-fit-arc.sh
+++ b/board/habanalabs/common/post-image-fit-arc.sh
@@ -70,7 +70,7 @@
 
 	hl_copy_keys ${BINS_DIR}
 
-	hl_copy_binaries ${BINS_DIR}
+	hl_copy_binaries ${BINS_PROD_DIR}
 
 	sed ${ITS_SED_PARAMS} ${ZEPHYR_ITS_TEMPLATE} > ${ZEPHYR_ITS_FILE}
 	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${ZEPHYR_ITS_FILE} -k ${STAGE_DIR} ${ZEPHYR_ITB_FILE}
diff --git a/board/habanalabs/common/post-image-fit-global.sh b/board/habanalabs/common/post-image-fit-global.sh
index f270c20..4d25876 100755
--- a/board/habanalabs/common/post-image-fit-global.sh
+++ b/board/habanalabs/common/post-image-fit-global.sh
@@ -175,7 +175,7 @@
 
 	hl_copy_keys ${BINS_DIR}
 
-	hl_copy_binaries ${BINS_DIR}
+	hl_copy_binaries ${BINS_PROD_DIR}
 
 	# Extract the first 4 bytes which contains the jump address
 	dd if=${BINS_DIR}/${HL_DEV}-zephyr-mgmt-iccm.bin.le of=${BINS_DIR}/${HL_DEV}-zephyr-mgmt-iccm.bin.jmp_sram bs=1 count=4
diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index 04b5213..c1ef89a 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -408,7 +408,7 @@
 
 pushd ${STAGE_DIR}
 
-	hl_copy_binaries ${BINS_DIR}
+	hl_copy_binaries ${BINS_PROD_DIR}
 
 	for i in $(seq 1 $NUM_PROD_KEYS); do
 		CUST=cust${i}
diff --git a/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh b/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
index c5b78c8..9e01a08 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-fit-arc.sh
@@ -55,7 +55,7 @@
 pushd ${STAGE_DIR}
 	hl_copy_keys ${BINS_DIR}
 
-	hl_copy_binaries ${BINS_DIR}
+	hl_copy_binaries ${BINS_PROD_DIR}
 
 	sed ${ITS_SED_PARAMS} ${ZEPHYR_ITS_TEMPLATE} > ${ZEPHYR_ITS_FILE}
 	${MKIMAGE} -D "-W no-unit_address_vs_reg" -f ${ZEPHYR_ITS_FILE} -k ${STAGE_DIR} ${ZEPHYR_ITB_FILE}
diff --git a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
index 8495399..64f4ae1 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
@@ -156,7 +156,7 @@
 
 pushd ${STAGE_DIR}
 
-	hl_copy_binaries ${BINS_DIR}
+	hl_copy_binaries ${BINS_PROD_DIR}
 
 	export IMG_PTR_SZ="$(get_struct_size $BOOT_POINTER_HEADER hl_fw_boot_ptrs)"
 	export IMG_DESC_SZ="$(get_struct_size $BOOT_POINTER_HEADER hl_fw_img_desc)"
diff --git a/board/habanalabs/gaudi3/post-image-fit.sh b/board/habanalabs/gaudi3/post-image-fit.sh
index 5f7b585..f9b1c75 100644
--- a/board/habanalabs/gaudi3/post-image-fit.sh
+++ b/board/habanalabs/gaudi3/post-image-fit.sh
@@ -8,7 +8,7 @@
 
 ETH_PHY_FW_HEX_FILE=${DEV_BASE}/eth_phy_fw.hex
 ETH_PHY_FW_HASH_FILE=${DEV_BASE}/eth_phy_fw.md5
-ETH_PHY_FW_BIN_FILE=${BINS_DIR}/${DEV_NAME}-eth-phy.bin.le
+ETH_PHY_FW_BIN_FILE=${BINS_PROD_DIR}/${DEV_NAME}-eth-phy.bin.le
 
 hl_create_section_variables
 
@@ -60,6 +60,11 @@
 	cp ${STAGE_DIR}/${DEV_PREFIX}-zephyr-event-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-event-dccm.bin.le
 
 	prep_eth_phy_fw_binary ${ETH_PHY_FW_HEX_FILE} ${ETH_PHY_FW_BIN_FILE}
+	cp ${BINS_DIR}/${DEV_PREFIX}-security-arc.bin.be ${BINS_PROD_DIR}/${DEV_PREFIX}-security-arc.bin.be
+	cp ${BINS_DIR}/${DEV_PREFIX}-images-pointers.bin.be ${BINS_PROD_DIR}/${DEV_PREFIX}-images-pointers.bin.be
+	cp ${BINS_DIR}/${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.be ${BINS_PROD_DIR}/${DEV_PREFIX}-ppboot-pcie_phy_fw.bin.be
+	cp ${BINS_DIR}/${DEV_PREFIX}-preboot.bin.be ${BINS_PROD_DIR}/${DEV_PREFIX}-preboot.bin.be
+	cp ${BINS_DIR}/${DEV_PREFIX}-erom.bin.be ${BINS_PROD_DIR}/${DEV_PREFIX}-erom.bin.be
 
 	return
 }
diff --git a/board/habanalabs/gaudi3_arc/post-image-agents.sh b/board/habanalabs/gaudi3_arc/post-image-agents.sh
index 06f41e1..7117224 100644
--- a/board/habanalabs/gaudi3_arc/post-image-agents.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-agents.sh
@@ -39,10 +39,10 @@
 ITS_SED_PARAMS+="s/@PREBOOT_BIN@/preboot/g;"
 ITS_SED_PARAMS+="s/@FW_LOADER_CONF_LOADABLES@/${LOADABLES}/g;"
 
-AGENT_NIC_FW_FILE=${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_NAME}-eth-phy.bin.le
-AGENT_SERDES_FW_NAME=zephyr_agentfw_serdes.bin
 
 source ${COMMON_BASE}/image-utility.sh
+AGENT_NIC_FW_FILE=${BINS_PROD_DIR}/${DEV_NAME}-eth-phy.bin.le
+AGENT_SERDES_FW_NAME=zephyr_agentfw_serdes.bin
 hl_create_section_variables
 
 ITS_SED_PARAMS+="s/@AGENT_FW_ADDR_HI@/${NIC_PHY_FW_ADDR_HI}/g;"
